torch/csrc/jit/script/compiler.cpp

```diff
@@ -826,14 +844,33 @@ struct to_ir {
     emit_if_expr(true_block, expr.true_expr());
     emit_if_expr(false_block, expr.false_expr());
 
+    auto true_type = unshapedType(true_block->outputs().at(0)->type());
+    auto false_type = unshapedType(false_block->outputs().at(0)->type());
+    if (*true_type != *false_type) {
```

Comment:
I don't get it, shouldn't we just assert they're `Int`s, or whatever we use to represent `bool`eans?

Reply:
`a > b : c`
These are the types of `b` and `c` not the type of `a`. We really should have a `join` operator on types. For now I force it to be exactly the same type.

Reply:
You meant `a ? b : c` here, right?